### ISSUE TITLE
SWProxy.exe doesn't work anymore

### ISSUE THREAD

**Issue by gs3rr4** [lstern/SWProxy-plugins#59]
I can't start the latest SWProxy.exe. There is an error I can't rly read, because the consle window is closing so fast.. somethin with python27 dll missing?

I never had problems before.

**Comment by Imberbus** [comment]
Same issue here. The error is : http://imgur.com/a/jdXG4
Managed to screenshot it.
The filename (_MEIXXXX) seems to be generated randomly at launch.

**Comment by Imberbus** [comment]
Back home. Does not work on my workstation either.
So laptop : broken
workstation : broken
Feel free to ask for infos if you need more to solve the issue mate.
Thanks

**Comment by jbag1489** [comment]
did a little searching, not sure if this is related, but it seems to be giving a python error and has a suggested fix. 

https://productforums.google.com/forum/#!msg/drive/XqGGU8rGIAk/2oPlad9EDAAJ

Good luck figuring it out @Imberbus  and @gs3rr4

**Comment by gs3rr4** [comment]
Where does this Python DLL even come from? I don't have Python installed and SWProxy always worked before.

Is Python a new "have to"?

**Comment by Imberbus** [comment]
Nope. I looked a little bit into it and it seems that SWProxy has a "python embedded" code that generates Python27.dll in theory. 
see : https://docs.python.org/2/faq/windows.html#id7 ( title : How can I embed Python into a Windows application?)
EDIT : or maybe not. I may have misunderstood
EDIT 2 : My bad, thats not it. Well I don't know where Python27.dll comes from. But when I run the old version of SWProxy, it does generated a directory in the Temp folder with various things in it, and python27.dll is one of them. 
So how and why is the old version able to generate and use python27.dll and the new one is not ? That is the real question.

**Comment by Imberbus** [comment]
BTW @Jbag1489 I tryed the permission change, it didn't fix it.

**Comment by jbag1489** [comment]
https://github.com/kakaroto/SWProxy/releases

I would grab the latest release from there, and then just move the plugins folder from this release into the folder from the other one, if that makes sense. 

I haven't tried updating or anything lately, but I have grabbed the plugins from here and dropped in my folder I already had from the above link

**Comment by Imberbus** [comment]
@Jbag1489 This does work for a part of the plugin. But for some plugin it doesn't seem to work (i.e SwarfarmLogger.py) 
error in the proxy.log : 
`2016-11-04 14:47:06,278: yapsy - ERROR - Unable to import plugin: F:\SWProxy\SWProxy-windows\plugins\SwarfarmLogger
Traceback (most recent call last):
  File "site-packages\yapsy\PluginManager.py", line 488, in loadPlugins
  File "F:\SWProxy\SWProxy-windows\plugins\SwarfarmLogger.py", line 7, in <module>
    import urllib2
ImportError: No module named urllib2`

**Comment by jbag1489** [comment]
@Imberbus I'm at work now, and can't try running anything, have you tried the latest full release from this project? (https://github.com/lstern/SWProxy-plugins/releases/download/0.101/SWProxy-windows.zip)

Seems that the problem is due to urllib2 not being available, which from what I can tell is a default python module. Do you have python installed?

If not, would you mind trying to install it and see if that helps anything? If you want to try installing python, here's a link: https://www.python.org/ftp/python/2.7.12/python-2.7.12.msi for the windows installer

**Comment by gs3rr4** [comment]
I tried the latest full release. That's the release where this error came from.

I installed Python now. Same error.

**Comment by jbag1489** [comment]
@porksmash Do you have any ideas as to what could be causing this? Just tagging you due to an apparent error with a SwarfarmLogger.py error

@gs3rr4 @Imberbus I'm out of ideas right now, especially not being able to test things right now.

**Comment by PeteAndersen** [comment]
I'm aware of the issue with swarfarmlogger plugin and the original swproxy releases. I'm going to rewrite the plugin to work with the available urllib module to alleviate these issues. It's a bit of a mystery why kakaroto's releases don't include urllib2 but this project does. 

I tried to rebuild the release from the latest sources and ran into several issues. I think we'll have to wait for @lstern to chime in since he's got the experience with releasing. I know he's been having a busy life lately.

**Comment by jbag1489** [comment]
Wish I knew more python to actually be helpful, but I'm working on it! I'll
get there eventually.

**Comment by Nightreaver** [comment]
So the source files run fine on windows - if you have python installed on windows just run the proxy from console @Jbag1489 @gs3rr4 

ie create a batch file "run.bat"
```
python.exe "%~dp0SWProxy.py"
```
and place it in the root folder where SWProxy.py is located (make sure it's a .bat not .txt or anything else)

if you have python not in your path variable you can use like (wherever your python.exe is located)
```
"C:\Python\Python27_x64\python.exe" "%~dp0SWProxy.py"
```

works fine for me... I think I switched to sources instantly because I didn't want to mess around with compiled .exe :D

**Comment by fperegrinvs** [comment]
fixed - see #68
